kroxylicious/src/main/java/io/kroxylicious/proxy/internal/package-info.java

```diff
@@ -0,0 +1,68 @@
+ * backend server (we read responses from the channel). So Filters are installed into the pipeline in the reverse order that they are declared in the Kroxylicious configuration
+ * YAML. Pipeline Handlers are invoked last-to-first for Requests, and first-to-last for Responses.
+ * <h3>KafkaProxyBackendHandler</h3>
+ * {@link io.kroxylicious.proxy.internal.KafkaProxyBackendHandler} signals to the Frontend Handler when the upstream channel is ready to be written to, and it writes
+ * Responses to the downstream channel after it has read them from the upstream channel.
+ * <h3>RPC Example</h3>
+ * For a session where a Client connects to Kroxylicious and makes a single DescribeCluster request:
+ * <ol>
+ *     <li>The downstream channel is initialized by {@link io.kroxylicious.proxy.internal.KafkaProxyInitializer}</li>
+ *     <li>The first message flows into KafkaRequestDecoder which decodes the bytes into a Request Frame</li>
+ *     <li>The Request Frame eventually flows into a {@link io.kroxylicious.proxy.internal.KafkaProxyFrontendHandler}</li>
+ *     <li>The Frontend Handler stores the Request Frame and calls selectServer on it's {@link io.kroxylicious.proxy.filter.NetFilter}</li>
+ *     <li>The NetFilter selects a backend server and calls initiateConnect on Frontend Handler</li>
+ *     <li>Frontend Handler establishes the upstream channel and installs handlers into it's pipeline including {@link io.kroxylicious.proxy.internal.KafkaProxyBackendHandler}</li>
+ *     <li>Backend Handler signals to Frontend Handler when it is active (ready to have bytes written to it)</li>
+ *     <li>Frontend Handler writes the stored Request Frame to upstream channel</li>
+ *     <li>Request Frame flows through user defined Custom Protocol Filters in the upstream channel</li>
+ *     <li>Request Frame flows through KafkaRequestEncoder (it also rewrites the correlationId and remembers the API and version of the RPC for that correlationId) and is eventually written to socket and transmitted to backend server</li>
+ *     <li>Backend server responds</li>
+ *     <li>Upstream channel reads response bytes from socket</li>
+ *     <li>Response bytes flow into KafkaResponseDecoder which looks up the API and version stored for the correlationId and uses that to decode the bytes into a Response Frame</li>
+ *     <li>Response Frame flows through user defined Custom Protocol Filters in the upstream channel</li>
+ *     <li>Response Frame eventually flows to {@link io.kroxylicious.proxy.internal.KafkaProxyBackendHandler} which then writes the Response to downstream channel</li>
+ *     <li>Response Frame flows through downstream channel in the outbound direction</li>
+ *     <li>Response Frame eventually flows to KafkaResponseEncoder where it is converted to bytes and is eventually written to the socket</li>
+ *     <li>Client reads bytes and is happy</li>
+ * </ol>
```

Comment:
I worry this is overly detailed and specific and I feel its very likely to drift from reality. 

So I'm +0 on this specific list.

Reply:
Yeah I think it will probably rot too easily, will drop it